File: sample/src/main/java/jp/co/pluswing/mysqlsample/sample/payload/AddCommentPayload.java
package jp.co.pluswing.mysqlsample.sample.payload;

import jp.co.pluswing.mysqlsample.sample.model.Comment;

public class AddCommentPayload {
    private AddCommentPayload(Boolean success, Comment comment) {
        this.success = success;
        this.comment = comment;
    }

    public final Boolean success;
    public final Comment comment;

    public static AddCommentPayload fail() {
        return new AddCommentPayload(false, null);
    }

    public static AddCommentPayload success(Comment comment) {
        return new AddCommentPayload(true, comment);
    }
}

File: sample/src/main/java/jp/co/pluswing/mysqlsample/sample/payload/AddCommentPayload.kt
package jp.co.pluswing.mysqlsample.sample.payload

import jp.co.pluswing.mysqlsample.sample.model.Comment

data class AddCommentPayload private constructor(val success: Boolean, val comment: Comment?) {
    companion object {
        fun fail(): AddCommentPayload {
            return AddCommentPayload(false, null)
        }

        fun success(comment: Comment): AddCommentPayload {
            return AddCommentPayload(true, comment)
        }
    }
}